src/Controller/SentencesController.php

```diff
@@ -555,7 +555,14 @@ public function search()
     public function advanced_search() {
         $search = new SentencesSearchForm();
 
-        $search->setData([]);
+        $data = $this->request->getParam('?');
+
+        if ($data) {
+            $search->setData($data);
+        } else {
+            $search->setData([]);
+        }
```

Comment:
I think this code can be simplified. There is another way to get the query parameters: [getQueryParams()](https://api.cakephp.org/3.7/class-Cake.Http.ServerRequest.html#_getQueryParams). This method will return an empty array if there are no parameters.

Reply:
Ah was thinking that too. Will fix.